server/util/fastcopy/fastcopy_windows.go

```diff
@@ -2,16 +2,174 @@
 
 package fastcopy
 
-import "os"
+import (
+	"fmt"
+	"os"
+	"syscall"
+	"unsafe"
 
+	"golang.org/x/sys/windows"
+)
+
+const (
+	// 1 GiB = 1024 MiB = 1024 * 1024 KiB = 1024 * 1024 * 1024 B
+	GiB = 1 << 30
+)
+
+var (
+	modkernel32   = syscall.NewLazyDLL("kernel32.dll")
+	procCopyFileW = modkernel32.NewProc("CopyFileW")
+
+	// ReFS only supports 64KiB and 4KiB cluster.
+	ReFSClusterSize = []int64{64 * 1024, 4 * 1024}
+)
+
+// User can set executor.enable_fastcopy_reflinking=true to force
+// using block cloning on Windows.
+// Otherwise, fallback to using CopyFileW which will use block block-cloning
+// automatically based on the underlying file system and Windows version.
```

Comment:
I don't fully understand the conditions under which `reflink` is faster than `FastCopy`. Is `reflink` supported on all file systems and Windows versions but `FastCopy` is only fast on devdrives and with new versions? Do we expect folks to use older versions of Windows for executors?

Reply:
Is it worth maintaining reflink in that case? If users want optimal performance, updating Windows to the latest version is probably what they'll do anyway. 

But the code and tests still look good (and we actually cover both cases), so I'll leave this decision to you. :⁠-⁠)